packages/dds/merge-tree/src/segmentPropertiesManager.ts

```diff
@@ -3,13 +3,16 @@
  * Licensed under the MIT License.
```

Comment:
this file contains most of the important changes, and is essentially a rewrite. i recommend reviewing in split view, rather than inline diff, as the changes are not incremental, so inline diffs will not make very much sense.